File: src/p0771/cpp/solution.cpp
class Solution {
public:
    int numJewelsInStones(const string &J, const string &S) {
        int result = 0;
        vector<int> dict(256, 0);
        for (auto k : J) {
            dict[k] = 1;
        }
        for (auto k : S) {
            result += dict[k];
        }
        return result;
    }
};

File: src/p0771/java/solution.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        int result = 0;
        int[] dict = new int[256];
        for (int i = 0; i < J.length(); i++) {
            dict[J.charAt(i)] = 1;
        }
        for (int i = 0; i < S.length(); i++) {
            result += dict[S.charAt(i)];
        }
        return result;
    }
}
